tests/tests-cube-sdcols.R

```diff
@@ -0,0 +1,21 @@
+# tests/tests-cube-sdcols.R
+# Test: cube() support for patterns with .SDcols
+# Related issue: https://github.com/Rdatatable/data.table/issues/7354
+
+library(data.table)
+
+cat("Running test for cube() with .SDcols pattern...\n")
+
+DT <- data.table(a1 = 1:2, a2 = 3:4, b = 5:6)
+
+result <- tryCatch({
+  cube(DT, .SDcols = patterns("^a"))
+  "No error"
```

Comment:
please move this test into inst/tests.Rraw

and test the output instead of ignoring the output.